tensorflow/python/data/experimental/kernel_tests/optimization/BUILD

```diff
@@ -26,6 +26,29 @@ tf_py_test(
     ],
 )
 
+tf_py_test(
+    name = "choose_fastest_branch_dataset_test",
```

Comment:
The tests in the newly introduced test target are timing out internally. Can you try manually running the test (after removing the no_oss tag?

Reply:
@jsimsa the test target is passing!
```
INFO: Elapsed time: 59.648s, Critical Path: 42.48s
INFO: 347 processes: 130 internal, 217 local.
INFO: Build completed successfully, 347 total actions
//tensorflow/python/data/experimental/kernel_tests/optimization:choose_fastest_branch_dataset_test PASSED in 22.2s
```

Reply:
Please change the size of the test target to "medium" it routinely takes >60 seconds to run internally, which causes it to time out (because its size is currently "small", which implies 60 second timeout).

Reply:
@jsimsa I made the change.